### PR TITLE
ipatests: Test cases for healthcheck File checker(s)

### PR DESCRIPTION
**Pull request by rcritten**
These check for modified file ownership (user and group) and 
too permissive and restrictive permissions across the three
types of files checked by the healthcheck FileCheck.
 
This replaces an existing test for TomcatFileCheck which adds
more functionality and consolidates all file checks together.

### CODE REVIEW

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -904,3 +835,330 @@ def test_ipahealthcheck_trust_agent_member_check(self):
...
+    """Fixture to change owner, group and/or mode
+
+       This can run against multiple files at once but only one host.
+    """
+
+    state = dict()
+
+    def _modify_permission(host, path, owner=None, group=None, mode=None):
+        """Change the ownership or mode of a path
+        """
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py:851]: ending `"""` should be on the same line for single line doc string as per pep8 standards.

Comment (rcritten) [reply ipatests/test_integration/test_ipahealthcheck.py]: fixed.

**Comment by rcritten** [comment]
Temp commit removed

**Comment by rcritten** [comment]
Rebased to force new run of CI.

**Comment by flo-renaud** [comment]
master:

* 550fbc0b9ff682d15a4f4a89974d0e506430fa8e ipatests: Test cases for healthcheck File checker(s)

**Pull request closed without merging**